### ISSUE TITLE
Help: Cannot create release build

### ISSUE THREAD

**Issue by NeverGivinUp** [koute/stdweb#368]
I can create a development build fine. But creating a release build results in the below error.

I am building on Windows 10. The error remains after I delete Cargo.lock. I cannot build in stable mode, because I need futures. I thought, maybe I am using incompatible versions of some library in some dependencies, but the dependency graph does not show anything possibly related to __web_malloc -- only rand, syn and quote have different versions.

What can I do?

```
cargo.exe +nightly web build --release --verbose
[Some successful compilation elided]

Running `rustc --edition=2018 --crate-name svg_routing_single_level src\main.rs --error-format json --crate-type bin --emit=dep-info,link -C opt-level=3 -C lto -C metadata=433246d159b61fab --out-dir F:\Softwareentwicklung\rust_projects\svg_routing_single_level\target\wasm32-unknown-unknown\release\deps --target wasm32-unknown-unknown -L dependency=F:\Softwareentwicklung\rust_projects\svg_routing_single_level\target\wasm32-unknown-unknown\release\deps -L dependency=F:\Softwareentwicklung\rust_projects\svg_routing_single_level\target\release\deps --extern alga=F:\Softwareentwicklung\rust_projects\svg_routing_single_level\target\wasm32-unknown-unknown\release\deps\libalga-31558a860e949929.rlib --extern debug_utils=F:\Softwareentwicklung\rust_projects\svg_routing_single_level\target\wasm32-unknown-unknown\release\deps\libdebug_utils-222345b62fc95ae9.rlib --extern decorum=F:\Softwareentwicklung\rust_projects\svg_routing_single_level\target\wasm32-unknown-unknown\release\deps\libdecorum-1d360c74e0f81370.rlib --extern futures=F:\Softwareentwicklung\rust_projects\svg_routing_single_level\target\wasm32-unknown-unknown\release\deps\libfutures-aa51d6aeb7ff4ce3.rlib --extern graph_basics=F:\Softwareentwicklung\rust_projects\svg_routing_single_level\target\wasm32-unknown-unknown\release\deps\libgraph_basics-dedadb37d431d7c2.rlib --extern hobofan_stdweb_logger=F:\Softwareentwicklung\rust_projects\svg_routing_single_level\target\wasm32-unknown-unknown\release\deps\libhobofan_stdweb_logger-0916b92348460d05.rlib --extern lazy_static=F:\Softwareentwicklung\rust_projects\svg_routing_single_level\target\wasm32-unknown-unknown\release\deps\liblazy_static-b1e8fd3ac5fed786.rlib --extern log=F:\Softwareentwicklung\rust_projects\svg_routing_single_level\target\wasm32-unknown-unknown\release\deps\liblog-cca6ebe5ba454bd3.rlib --extern loop_stream=F:\Softwareentwicklung\rust_projects\svg_routing_single_level\target\wasm32-unknown-unknown\release\deps\libloop_stream-4cc5bffacdb1e48e.rlib --extern na=F:\Softwareentwicklung\rust_projects\svg_routing_single_level\target\wasm32-unknown-unknown\release\deps\libnalgebra-c8a214e0019ec2d5.rlib --extern num=F:\Softwareentwicklung\rust_projects\svg_routing_single_level\target\wasm32-unknown-unknown\release\deps\libnum-806c858ac8457025.rlib --extern rand=F:\Softwareentwicklung\rust_projects\svg_routing_single_level\target\wasm32-unknown-unknown\release\deps\librand-724170de2eda434d.rlib --extern serde=F:\Softwareentwicklung\rust_projects\svg_routing_single_level\target\wasm32-unknown-unknown\release\deps\libserde-b60d9034ca34dda0.rlib --extern serde_json=F:\Softwareentwicklung\rust_projects\svg_routing_single_level\target\wasm32-unknown-unknown\release\deps\libserde_json-e5e545cdcb10f3dc.rlib --extern single_level_force_directed_graph_layout=F:\Softwareentwicklung\rust_projects\svg_routing_single_level\target\wasm32-unknown-unknown\release\deps\libsingle_level_force_directed_graph_layout-153ab2543c5b6bd2.rlib --extern stdweb=F:\Softwareentwicklung\rust_projects\svg_routing_single_level\target\wasm32-unknown-unknown\release\deps\libstdweb-4ab77e3d09d58aa4.rlib --cfg cargo_web`
[Some unused warnings elided]

warning: Linking globals named '__web_malloc': symbol multiply defined!
error: failed to load bc of "stdweb.loqgzewm-cgu.1": 
error: aborting due to previous error
error: Could not compile `svg_routing_single_level`.

Caused by:
  process didn't exit successfully: `rustc --edition=2018 --crate-name svg_routing_single_level src\main.rs --error-format json --crate-type bin --emit=dep-info,link -C opt-level=3 -C lto -C metadata=433246d159b61fab --out-dir F:\Softwareentwicklung\rust_projects\svg_routing_single_level\target\wasm32-unknown-unknown\release\deps --target wasm32-unknown-unknown -L dependency=F:\Softwareentwicklung\rust_projects\svg_routing_single_level\target\wasm32-unknown-unknown\release\deps -L dependency=F:\Softwareentwicklung\rust_projects\svg_routing_single_level\target\release\deps --extern alga=F:\Softwareentwicklung\rust_projects\svg_routing_single_level\target\wasm32-unknown-unknown\release\deps\libalga-31558a860e949929.rlib --extern debug_utils=F:\Softwareentwicklung\rust_projects\svg_routing_single_level\target\wasm32-unknown-unknown\release\deps\libdebug_utils-222345b62fc95ae9.rlib --extern decorum=F:\Softwareentwicklung\rust_projects\svg_routing_single_level\target\wasm32-unknown-unknown\release\deps\libdecorum-1d360c74e0f81370.rlib --extern futures=F:\Softwareentwicklung\rust_projects\svg_routing_single_level\target\wasm32-unknown-unknown\release\deps\libfutures-aa51d6aeb7ff4ce3.rlib --extern graph_basics=F:\Softwareentwicklung\rust_projects\svg_routing_single_level\target\wasm32-unknown-unknown\release\deps\libgraph_basics-dedadb37d431d7c2.rlib --extern hobofan_stdweb_logger=F:\Softwareentwicklung\rust_projects\svg_routing_single_level\target\wasm32-unknown-unknown\release\deps\libhobofan_stdweb_logger-0916b92348460d05.rlib --extern lazy_static=F:\Softwareentwicklung\rust_projects\svg_routing_single_level\target\wasm32-unknown-unknown\release\deps\liblazy_static-b1e8fd3ac5fed786.rlib --extern log=F:\Softwareentwicklung\rust_projects\svg_routing_single_level\target\wasm32-unknown-unknown\release\deps\liblog-cca6ebe5ba454bd3.rlib --extern loop_stream=F:\Softwareentwicklung\rust_projects\svg_routing_single_level\target\wasm32-unknown-unknown\release\deps\libloop_stream-4cc5bffacdb1e48e.rlib --extern na=F:\Softwareentwicklung\rust_projects\svg_routing_single_level\target\wasm32-unknown-unknown\release\deps\libnalgebra-c8a214e0019ec2d5.rlib --extern num=F:\Softwareentwicklung\rust_projects\svg_routing_single_level\target\wasm32-unknown-unknown\release\deps\libnum-806c858ac8457025.rlib --extern rand=F:\Softwareentwicklung\rust_projects\svg_routing_single_level\target\wasm32-unknown-unknown\release\deps\librand-724170de2eda434d.rlib --extern serde=F:\Softwareentwicklung\rust_projects\svg_routing_single_level\target\wasm32-unknown-unknown\release\deps\libserde-b60d9034ca34dda0.rlib --extern serde_json=F:\Softwareentwicklung\rust_projects\svg_routing_single_level\target\wasm32-unknown-unknown\release\deps\libserde_json-e5e545cdcb10f3dc.rlib --extern single_level_force_directed_graph_layout=F:\Softwareentwicklung\rust_projects\svg_routing_single_level\target\wasm32-unknown-unknown\release\deps\libsingle_level_force_directed_graph_layout-153ab2543c5b6bd2.rlib --extern stdweb=F:\Softwareentwicklung\rust_projects\svg_routing_single_level\target\wasm32-unknown-unknown\release\deps\libstdweb-4ab77e3d09d58aa4.rlib --cfg cargo_web` (exit code: 1)
error: build failed

Process finished with exit code 101
```

**Comment by koute** [comment]
1. Are you using `stdweb` from `crates.io` or from git?
2. What is your dependency graph? (`cargo tree` output)
3. Does this happen only with your specific app or with any hello-world sized app?

**Comment by NeverGivinUp** [comment]
I was using a modified copy from git. The crates.io version does not compile with `futures-support`, because the futures API has changed.  Without `futures-support` I can build a release version.

I was able to create a minimal test case that reproduces my problem

```toml
[package]
name = "std_release_build_test"
version = "0.1.0"
edition = "2018"

[dependencies]
stdweb ={git="https://github.com/koute/stdweb.git", features = ["futures-support", "experimental_features_which_may_break_on_minor_version_bumps"]}
rand = { version = "0.7", features = ["stdweb"] }
```
```rust
#[macro_use]
extern crate stdweb;

use rand::Rng;

fn main() {
    stdweb::initialize();
    let rng = &mut rand::thread_rng();
    let message = format!("Hello, 世界!\nSay {}", rng.gen_range(0.0, 256.0));
    js! {
        alert( @{message} );
    }

    stdweb::event_loop();
}
```

Compilation results in 
```
cargo.exe +nightly web build --release
   Compiling std_release_build_test v0.1.0 (F:\Softwareentwicklung\rust_projects\std_release_build_test)
error: linking with `rust-lld` failed: exit code: 1
  |
  = note: "rust-lld" "-flavor" "wasm" "--no-threads" "-z" "stack-size=1048576" "--stack-first" "--allow-undefined" "--fatal-warnings" "--no-demangle" "--export-dynamic" "--no-entry" "-L" "C:\\Users\\Navid\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\wasm32-unknown-unknown\\lib" "F:\\Softwareentwicklung\\rust_projects\\std_release_build_test\\target\\wasm32-unknown-unknown\\release\\deps\\std_release_build_test.std_release_build_test.a600v91q-cgu.0.rcgu.o" "F:\\Softwareentwicklung\\rust_projects\\std_release_build_test\\target\\wasm32-unknown-unknown\\release\\deps\\std_release_build_test.std_release_build_test.a600v91q-cgu.1.rcgu.o" "F:\\Softwareentwicklung\\rust_projects\\std_release_build_test\\target\\wasm32-unknown-unknown\\release\\deps\\std_release_build_test.std_release_build_test.a600v91q-cgu.10.rcgu.o" "F:\\Softwareentwicklung\\rust_projects\\std_release_build_test\\target\\wasm32-unknown-unknown\\release\\deps\\std_release_build_test.std_release_build_test.a600v91q-cgu.11.rcgu.o" "F:\\Softwareentwicklung\\rust_projects\\std_release_build_test\\target\\wasm32-unknown-unknown\\release\\deps\\std_release_build_test.std_release_build_test.a600v91q-cgu.12.rcgu.o" "F:\\Softwareentwicklung\\rust_projects\\std_release_build_test\\target\\wasm32-unknown-unknown\\release\\deps\\std_release_build_test.std_release_build_test.a600v91q-cgu.13.rcgu.o" "F:\\Softwareentwicklung\\rust_projects\\std_release_build_test\\target\\wasm32-unknown-unknown\\release\\deps\\std_release_build_test.std_release_build_test.a600v91q-cgu.14.rcgu.o" "F:\\Softwareentwicklung\\rust_projects\\std_release_build_test\\target\\wasm32-unknown-unknown\\release\\deps\\std_release_build_test.std_release_build_test.a600v91q-cgu.15.rcgu.o" "F:\\Softwareentwicklung\\rust_projects\\std_release_build_test\\target\\wasm32-unknown-unknown\\release\\deps\\std_release_build_test.std_release_build_test.a600v91q-cgu.2.rcgu.o" "F:\\Softwareentwicklung\\rust_projects\\std_release_build_test\\target\\wasm32-unknown-unknown\\release\\deps\\std_release_build_test.std_release_build_test.a600v91q-cgu.3.rcgu.o" "F:\\Softwareentwicklung\\rust_projects\\std_release_build_test\\target\\wasm32-unknown-unknown\\release\\deps\\std_release_build_test.std_release_build_test.a600v91q-cgu.4.rcgu.o" "F:\\Softwareentwicklung\\rust_projects\\std_release_build_test\\target\\wasm32-unknown-unknown\\release\\deps\\std_release_build_test.std_release_build_test.a600v91q-cgu.5.rcgu.o" "F:\\Softwareentwicklung\\rust_projects\\std_release_build_test\\target\\wasm32-unknown-unknown\\release\\deps\\std_release_build_test.std_release_build_test.a600v91q-cgu.6.rcgu.o" "F:\\Softwareentwicklung\\rust_projects\\std_release_build_test\\target\\wasm32-unknown-unknown\\release\\deps\\std_release_build_test.std_release_build_test.a600v91q-cgu.7.rcgu.o" "F:\\Softwareentwicklung\\rust_projects\\std_release_build_test\\target\\wasm32-unknown-unknown\\release\\deps\\std_release_build_test.std_release_build_test.a600v91q-cgu.8.rcgu.o" "F:\\Softwareentwicklung\\rust_projects\\std_release_build_test\\target\\wasm32-unknown-unknown\\release\\deps\\std_release_build_test.std_release_build_test.a600v91q-cgu.9.rcgu.o" "-o" "F:\\Softwareentwicklung\\rust_projects\\std_release_build_test\\target\\wasm32-unknown-unknown\\release\\deps\\std_release_build_test.wasm" "--export" "main" "--export" "__web_malloc" "--export" "__web_free" "--export" "__web_malloc" "--export" "__web_free" "--export=__heap_base" "--export=__data_end" "F:\\Softwareentwicklung\\rust_projects\\std_release_build_test\\target\\wasm32-unknown-unknown\\release\\deps\\std_release_build_test.1oqlv4x85ht878ce.rcgu.o" "--gc-sections" "-O3" "-L" "F:\\Softwareentwicklung\\rust_projects\\std_release_build_test\\target\\wasm32-unknown-unknown\\release\\deps" "-L" "F:\\Softwareentwicklung\\rust_projects\\std_release_build_test\\target\\release\\deps" "-L" "C:\\Users\\Navid\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\wasm32-unknown-unknown\\lib" "F:\\Softwareentwicklung\\rust_projects\\std_release_build_test\\target\\wasm32-unknown-unknown\\release\\deps\\librand-9182c92282233832.rlib" "F:\\Softwareentwicklung\\rust_projects\\std_release_build_test\\target\\wasm32-unknown-unknown\\release\\deps\\librand_chacha-f51cfa78c643f099.rlib" "F:\\Softwareentwicklung\\rust_projects\\std_release_build_test\\target\\wasm32-unknown-unknown\\release\\deps\\libc2_chacha-ac5be2aa6066a9c8.rlib" "F:\\Softwareentwicklung\\rust_projects\\std_release_build_test\\target\\wasm32-unknown-unknown\\release\\deps\\libppv_lite86-0685a0a529b0ca76.rlib" "F:\\Softwareentwicklung\\rust_projects\\std_release_build_test\\target\\wasm32-unknown-unknown\\release\\deps\\liblazy_static-23dd9c3857571371.rlib" "F:\\Softwareentwicklung\\rust_projects\\std_release_build_test\\target\\wasm32-unknown-unknown\\release\\deps\\librand_core-ef44083616d3b9c9.rlib" "F:\\Softwareentwicklung\\rust_projects\\std_release_build_test\\target\\wasm32-unknown-unknown\\release\\deps\\libgetrandom-a61c041400faae9e.rlib" "F:\\Softwareentwicklung\\rust_projects\\std_release_build_test\\target\\wasm32-unknown-unknown\\release\\deps\\libstdweb-45d2db784f9c9e52.rlib" "F:\\Softwareentwicklung\\rust_projects\\std_release_build_test\\target\\wasm32-unknown-unknown\\release\\deps\\libstdweb_internal_runtime-a0b7d40e4f3260c7.rlib" "F:\\Softwareentwicklung\\rust_projects\\std_release_build_test\\target\\wasm32-unknown-unknown\\release\\deps\\libcfg_if-445d1612558e7dd5.rlib" "F:\\Softwareentwicklung\\rust_projects\\std_release_build_test\\target\\wasm32-unknown-unknown\\release\\deps\\libstdweb-b7cc1dd64022821a.rlib" "F:\\Softwareentwicklung\\rust_projects\\std_release_build_test\\target\\wasm32-unknown-unknown\\release\\deps\\libdiscard-5489fc9c1746fa14.rlib" "F:\\Softwareentwicklung\\rust_projects\\std_release_build_test\\target\\wasm32-unknown-unknown\\release\\deps\\libstdweb_internal_runtime-836781eb26dc5d22.rlib" "F:\\Softwareentwicklung\\rust_projects\\std_release_build_test\\target\\wasm32-unknown-unknown\\release\\deps\\libfutures_executor-63198a4bbe69a858.rlib" "F:\\Softwareentwicklung\\rust_projects\\std_release_build_test\\target\\wasm32-unknown-unknown\\release\\deps\\libnum_cpus-3a09c16807e53d58.rlib" "F:\\Softwareentwicklung\\rust_projects\\std_release_build_test\\target\\wasm32-unknown-unknown\\release\\deps\\liblibc-73954e9e31571642.rlib" "F:\\Softwareentwicklung\\rust_projects\\std_release_build_test\\target\\wasm32-unknown-unknown\\release\\deps\\libfutures_channel-d166b0ecd18ed89c.rlib" "F:\\Softwareentwicklung\\rust_projects\\std_release_build_test\\target\\wasm32-unknown-unknown\\release\\deps\\libfutures_util-8813dd9c4fd91712.rlib" "F:\\Softwareentwicklung\\rust_projects\\std_release_build_test\\target\\wasm32-unknown-unknown\\release\\deps\\libslab-49a364eeeadd96c7.rlib" "F:\\Softwareentwicklung\\rust_projects\\std_release_build_test\\target\\wasm32-unknown-unknown\\release\\deps\\libpin_utils-a037821269d4fc74.rlib" "F:\\Softwareentwicklung\\rust_projects\\std_release_build_test\\target\\wasm32-unknown-unknown\\release\\deps\\libfutures_core-ce52c82a8637bd50.rlib" "F:\\Softwareentwicklung\\rust_projects\\std_release_build_test\\target\\wasm32-unknown-unknown\\release\\deps\\libserde_json-e3c1e326ed1e7d2a.rlib" "F:\\Softwareentwicklung\\rust_projects\\std_release_build_test\\target\\wasm32-unknown-unknown\\release\\deps\\libryu-46b4acff1c2a49ed.rlib" "F:\\Softwareentwicklung\\rust_projects\\std_release_build_test\\target\\wasm32-unknown-unknown\\release\\deps\\libitoa-0f06e25c693c651e.rlib" "F:\\Softwareentwicklung\\rust_projects\\std_release_build_test\\target\\wasm32-unknown-unknown\\release\\deps\\libserde-0b0007fde9c62817.rlib" "C:\\Users\\Navid\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\wasm32-unknown-unknown\\lib\\libstd-6562854ef6152a02.rlib" "C:\\Users\\Navid\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\wasm32-unknown-unknown\\lib\\libpanic_abort-d671122d9898f1e2.rlib" "C:\\Users\\Navid\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\wasm32-unknown-unknown\\lib\\libdlmalloc-50ba03940f0d2091.rlib" "C:\\Users\\Navid\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\wasm32-unknown-unknown\\lib\\libbacktrace-457d39515c4aad54.rlib" "C:\\Users\\Navid\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\wasm32-unknown-unknown\\lib\\librustc_demangle-fb44312e4ba0e61e.rlib" "C:\\Users\\Navid\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\wasm32-unknown-unknown\\lib\\libhashbrown-21d5f2b110019126.rlib" "C:\\Users\\Navid\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\wasm32-unknown-unknown\\lib\\librustc_std_workspace_alloc-cfcf31d57497969d.rlib" "C:\\Users\\Navid\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\wasm32-unknown-unknown\\lib\\libunwind-2659d4aef9482483.rlib" "C:\\Users\\Navid\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\wasm32-unknown-unknown\\lib\\libcfg_if-d2d4ea0a1f62f42e.rlib" "C:\\Users\\Navid\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\wasm32-unknown-unknown\\lib\\liblibc-a074862ad51f65ae.rlib" "C:\\Users\\Navid\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\wasm32-unknown-unknown\\lib\\liballoc-290decfd941290c6.rlib" "C:\\Users\\Navid\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\wasm32-unknown-unknown\\lib\\librustc_std_workspace_core-223fcd36a452bbca.rlib" "C:\\Users\\Navid\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\wasm32-unknown-unknown\\lib\\libcore-10d3699fc15530b7.rlib" "C:\\Users\\Navid\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\wasm32-unknown-unknown\\lib\\libcompiler_builtins-0ac06e0f42bf50f5.rlib"
  = note: rust-lld: error: duplicate symbol: __web_malloc
          >>> defined in F:\Softwareentwicklung\rust_projects\std_release_build_test\target\wasm32-unknown-unknown\release\deps\libstdweb-45d2db784f9c9e52.rlib(stdweb-45d2db784f9c9e52.stdweb.97by1qzw-cgu.1.rcgu.o)
          >>> defined in F:\Softwareentwicklung\rust_projects\std_release_build_test\target\wasm32-unknown-unknown\release\deps\libstdweb-b7cc1dd64022821a.rlib(stdweb-b7cc1dd64022821a.stdweb.568pdib2-cgu.5.rcgu.o)
          
          rust-lld: error: duplicate symbol: __web_free
          >>> defined in F:\Softwareentwicklung\rust_projects\std_release_build_test\target\wasm32-unknown-unknown\release\deps\libstdweb-45d2db784f9c9e52.rlib(stdweb-45d2db784f9c9e52.stdweb.97by1qzw-cgu.1.rcgu.o)
          >>> defined in F:\Softwareentwicklung\rust_projects\std_release_build_test\target\wasm32-unknown-unknown\release\deps\libstdweb-b7cc1dd64022821a.rlib(stdweb-b7cc1dd64022821a.stdweb.568pdib2-cgu.5.rcgu.o)
          

error: aborting due to previous error
error: Could not compile `std_release_build_test`.

To learn more, run the command again with --verbose.
error: build failed

Process finished with exit code 101
```

**Comment by NeverGivinUp** [comment]
@koute have you been able to reproduce my problem? Do you know (or have a guess) as to its source?

**Comment by koute** [comment]
Sorry for the delay.

Yes. The problem is the fact that you have two `stdweb`s simultaneously in your crate graph. (If you want to do that in the future I suggest [you read this](https://doc.rust-lang.org/edition-guide/rust-2018/cargo-and-crates-io/replacing-dependencies-with-patch.html).)

I just released 0.4.19, so you shouldn't have to use the git version anymore.

**Comment by NeverGivinUp** [comment]
Patching solved it for me.
Thanks :)
